app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,42 @@
+class CommentsController < ApplicationController
+  def new
+    @comment = Comment.new
+    @user = current_user
+    @post = Post.first
+    respond_to do |format|
+      format.html { render :new, locals: { comment: @comment } }
+    end
+  end
+
+  def create
+    # new object from params
+    @comment = Comment.new(post_params)
+
+    # respond_to block
+    respond_to do |format|
+      format.html do
+        if @comment.save
+          # success message
+          post = Post.find(post_params[:author_id])
+          Comment.update_post_comments_count(post)
+          flash[:success] = 'Comment created successfully'
+          # redirect to index
+          redirect_to user_posts_url
+        else
+          flash.now[:error] = 'Error: Comment could not be created'
+          # render new
+          render :new, locals: { comment: @comment }
+        end
+      end
+    end
+  end
+
+  private
+
+  def post_params
+    post_params = params.require(:comment).permit(:text)
+    post_params[:author_id] = current_user.id
+    post_params[:post_id] = Post.first.id
+    post_params
+  end
```

Comment:
- [ ] Great work trying to implement the change requested from a previous reviewer. I noticed that all the comments I create are for the first post only. I should be able to comment on any post I like. To implement this, I would advise you to do the following:
- Move the **_Add Comment_** button to the `post#show` page so that you can have access to the id for that particular post.
- Find the particular post by the id of the post passed in as params and then assign the id of the post to `post_params[:post_id]`.

Reply:
Thanks for the review

Reply:
@jssol, you are welcome. 👍